In [5]:
import pandas as pd

In [6]:
TFL = pd.read_csv('data/tfl_journeys_cleaned.csv')

In [7]:
%%sql
SELECT 
    YEAR,
    SUM(JOURNEYS_MILLIONS) AS Total_Journeys_Millions
FROM 
    TFL
GROUP BY 
    YEAR
ORDER BY 
    YEAR;

year  Total_Journeys_Millions
0   2010              1447.306835
1   2011              1588.567597
2   2012              1520.491438
3   2013              1559.900001
4   2014              1770.023002
5   2015              1907.196744
6   2016              1932.319081
7   2017              1907.690360
8   2018              2093.539708
9   2019              2250.906403
10  2020              1186.590978
11  2021              2283.443160
12  2022              2995.516679

In [15]:
%%sql
SELECT 
    YEAR,
    JOURNEY_TYPE,
    SUM(JOURNEYS_MILLIONS) AS Total_Journeys_Millions
FROM 
    TFL
GROUP BY 
    YEAR, JOURNEY_TYPE
ORDER BY 
    YEAR, Total_Journeys_Millions DESC
LIMIT 12;

year       journey_type  Total_Journeys_Millions
0   2010  Underground & DLR              1096.145589
1   2010                Bus               300.474666
2   2010               Tram                25.733162
3   2010         Overground                24.953418
4   2011  Underground & DLR              1156.647655
5   2011                Bus               312.257109
6   2011         Overground                93.467759
7   2011               Tram                26.195074
8   2012  Underground & DLR              1225.149122
9   2012                Bus               151.936327
10  2012         Overground               113.816793
11  2012               Tram                27.673344

In [16]:
%%sql
SELECT 
    MONTH,
    AVG(JOURNEYS_MILLIONS) AS Average_Journeys_Millions
FROM 
    TFL
GROUP BY 
    MONTH
ORDER BY 
    MONTH;

month  Average_Journeys_Millions
0       1                  32.280276
1       2                  28.664966
2       3                  28.634722
3       4                  29.826973
4       5                  41.781839
5       6                  33.362790
6       7                  32.376643
7       8                  31.876057
8       9                  32.670364
9      10                  41.833104
10     11                  29.188880
11     12                  30.070191

### Trend Analysis Findings

#### Overview of Total Journeys

The total volume of journeys has shown a general upward trend over the years, with some fluctuations:

- **2010**: 1,447.31 million journeys
- **2011**: 1,588.57 million journeys
- **2012**: 1,520.49 million journeys
- **2013**: 1,559.90 million journeys
- **2014**: 1,770.02 million journeys
- **2015**: 1,907.20 million journeys
- **2016**: 1,932.32 million journeys
- **2017**: 1,907.69 million journeys
- **2018**: 2,093.54 million journeys
- **2019**: 2,250.91 million journeys
- **2020**: 1,186.59 million journeys
- **2021**: 2,283.44 million journeys
- **2022**: 2,995.52 million journeys

Notable observations include a steady increase from 2010 to 2019, a significant decrease in 2020, likely due to the COVID-19 pandemic, and a remarkable increase in 2022, reaching the highest total volume of journeys in the dataset.

### Conclusions

London's public transport system has experienced significant growth over the past decade, reflecting the city's increasing population and reliance on efficient transportation. The steady rise in journey volume demonstrates the importance of TfL's services in facilitating mobility and connectivity for Londoners.

The decrease in journey volume in 2020 highlights the impact of external factors such as the COVID-19 pandemic, which led to travel restrictions and reduced mobility. However, the subsequent rebound in 2021 and 2022 signals resilience and adaptability within London's transport network, as well as the gradual return to pre-pandemic travel patterns.

Overall, the data underscores the vital role of public transport in supporting London's dynamic urban environment, contributing to economic growth, sustainability, and quality of life for residents and visitors alike.